File: domain/src/testFixtures/java/com/study/platform/support/fake/FakeRedisMessagePublisher.java
package com.study.platform.support.fake;

import com.study.platform.global.redis.RedisMessagePublisher;

import java.util.ArrayList;
import java.util.List;

public class FakeRedisMessagePublisher implements RedisMessagePublisher {

    private final List<String> publishedChannels = new ArrayList<>();

    @Override
    public void publish(String channel, String message) {
        publishedChannels.add(channel);
    }

    public boolean wasPublishedTo(String channel) {
        return publishedChannels.contains(channel);
    }

    public List<String> getPublishedChannels() {
        return publishedChannels;
    }
}

File: domain/src/testFixtures/kotlin/com/study/platform/support/fake/FakeRedisMessagePublisher.kt
package com.study.platform.support.fake

import com.study.platform.global.redis.RedisMessagePublisher

class FakeRedisMessagePublisher : RedisMessagePublisher {

    private val publishedChannels: MutableList<String> = ArrayList()

    override fun publish(channel: String, message: String) {
        publishedChannels.add(channel)
    }

    fun wasPublishedTo(channel: String): Boolean = publishedChannels.contains(channel)

    fun getPublishedChannels(): List<String> = publishedChannels
}
